File: leet-code-solutions/check-if-number-is-a-sum-of-powers-of-three.py
class Solution:
    def checkPowersOfThree(self, n: int) -> bool:
        index = 15
        while index >= 0:
            num = 3 ** index
            if(num <= n):
                n = n - num
            if(n == 0):
                return True
            index -= 1
        return False            

File: leet-code-solutions/check-if-number-is-a-sum-of-powers-of-three.cpp
class Solution {
public:
    bool checkPowersOfThree(int n) {
        for(int index = 15; index >= 0; index--){
            int num = pow(3, index);
            if(num <= n)
                n -= num;
            if(n == 0) return true;
        }
        return false;
    }
};